x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportPreviewTransformAction.java

```diff
@@ -119,9 +128,27 @@ protected TransportPreviewTransformAction(
     }
 
     @Override
-    protected void doExecute(Task task, PreviewTransformAction.Request request, ActionListener<PreviewTransformAction.Response> listener) {
+    protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
         final ClusterState clusterState = clusterService.state();
         TransformNodes.warnIfNoTransformNodes(clusterState);
+        final DiscoveryNodes nodes = clusterState.nodes();
+
+        // Redirection can only be performed between nodes that are at least 7.13.
+        if (nodes.getMinNodeVersion().onOrAfter(Version.V_7_13_0)
+                && request.getConfig().getSource().requiresRemoteCluster()
+                && (isRemoteClusterClientNode == false)) {
```

Comment:
I think it would be simpler and better for the future to always redirect preview to a transform node, that means checking if _this_ node is sufficient(transform+remote_cluster_if_needed) and redirect using `selectAnyTransformRemoteNode` if not.